PyPDF2/_merger.py

```diff
@@ -78,17 +78,29 @@ class PdfMerger:
     :param bool strict: Determines whether user should be warned of all
             problems and also causes some correctable problems to be fatal.
             Defaults to ``False``.
+    :param fileobj: Output file. Can be a filename or any kind of
+            file-like object.
     """
 
-    def __init__(self, strict: bool = False) -> None:
+    def __init__(self, fileobj: StrByteType = "", strict: bool = False) -> None:
         self.inputs: List[Tuple[Any, PdfReader, bool]] = []
         self.pages: List[Any] = []
         self.output: Optional[PdfWriter] = PdfWriter()
         self.bookmarks: OutlinesType = []
         self.named_dests: List[Any] = []
         self.id_count = 0
+        self.fileobj = fileobj
         self.strict = strict
 
+    # There is nothing to do.
+    def __enter__(self):
+        return self
+
+    # Write to the fileobj and close the merger.
+    def __exit__(self, *args):
```

Comment:
I'm not sure if mypy likes the variable args, so may need to list all the separate args, something like:
```
    def __exit__(
        self, exc_type: Optional[Type[BaseException]], exc: Optional[BaseException], traceback: 
    Optional[TracebackType]
    ) -> None: ...
```